lib/EnsEMBL/REST/Controller/Info.pm

```diff
@@ -331,6 +331,131 @@ sub consequence_types_GET {
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  my $expand = $c->request->param('expand');
+  $c->log()->info("Retrieving information about all genomes");
+  my $info = $lookup->adaptor()->fetch_by_any_name($name);
+  $c->go( 'ReturnError', 'custom', ["Genome $name not found"] )
+  unless defined $info;
+  $self->status_ok( $c, entity => _expand_genome($info, $expand) );
+  return;
+}
+
+sub divisions : Chained('/') PathPart('info/divisions') :
+  ActionClass('REST') : Args(0) { }
+
+sub divisions_GET {
+  my ( $self, $c, $division ) = @_;
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  $c->log()->info("Retrieving list of divisions");
+  my $divs = $lookup->adaptor()->list_divisions();
+  $self->status_ok( $c, entity => $divs );
+  return;
+}
+
+sub genomes_division : Chained('/') PathPart('info/genomes/division') :
+  ActionClass('REST') : Args(1) { }
+
+sub genomes_division_GET {
+  my ( $self, $c, $division ) = @_;
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  my $expand = $c->request->param('expand');
+  $c->log()
+  ->info(
+    "Retrieving information about genomes from division $division");
+  my @infos = map { _expand_genome($_, $expand) }
+  @{ $lookup->adaptor()->fetch_all_by_division($division) };
+  $self->status_ok( $c, entity => \@infos );
+  return;
+}
+
+sub genomes_assembly : Chained('/') PathPart('info/genomes/assembly') :
+  ActionClass('REST') : Args(1) { }
+
+sub genomes_assembly_GET {
+  my ( $self, $c, $acc ) = @_;
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  my $expand = $c->request->param('expand');
+  my $info = $lookup->adaptor()->fetch_by_assembly_accession($acc);
+  $c->go( 'ReturnError', 'custom',
+      ["Genome with assembly accession $acc not found"] )
+  unless defined $info;
+  $self->status_ok( $c, entity => _expand_genome($info, $expand));
+  return;
+}
+
+sub genomes_accession : Chained('/') PathPart('info/genomes/accession')
+  : ActionClass('REST') : Args(1) { }
+
+sub genomes_accession_GET {
+  my ( $self, $c, $acc ) = @_;
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  my $expand = $c->request->param('expand');
+  my @infos  = map { _expand_genome($_, $expand) }
+  @{ $lookup->adaptor()->fetch_all_by_sequence_accession($acc) };
+  $self->status_ok( $c, entity => \@infos );
+  return;
+}
+
+sub genomes_taxonomy : Chained('/') PathPart('info/genomes/taxonomy') :
+  ActionClass('REST') : Args(1) { }
+
+sub genomes_taxonomy_GET {
+  my ( $self, $c, $taxon ) = @_;
+  # lazy load the registry
+  $c->model('Registry')->_registry();
+  my $lookup = $c->model('Registry')->_lookup();
+  my $expand = $c->request->param('expand');
+  $c->log()
```

Comment:
Indent failure. ->info and @{ } could be formatted to more clearly with respect to the run-on from the previous line.

Reply:
Reformatted